packages/maskbook/src/plugins/Trader/UI/trending/SearchResultView.tsx

```diff
@@ -154,14 +156,9 @@ export function SearchResultView(props: SearchResultViewProps) {
     const canSwap = trending.coin.eth_address || trending.coin.symbol.toLowerCase() === 'eth'
     const swapTabIndex = dataProvider !== DataProvider.UNISWAP ? 3 : 1
     const fromToken = chainId === ChainId.Mainnet && coin.is_mirrored ? UST : createEtherToken(chainId)
+    const { decimals } = coinDetailed ?? trending.coin
```

Comment:
Maybe reduce this local variable.

Reply:
Does `createERC20Token()` compute the decimals out?

```
export function createERC20Token(chainId: ChainId, address: string, decimals: number, name: string, symbol: string) {
    return {
        type: EthereumTokenType.ERC20,
        chainId,
        address,
        decimals,
        name,
        symbol,
    } as ERC20TokenDetailed
}
```